nomos-da/network/core/src/swarm/validator.rs

```diff
@@ -102,7 +104,17 @@ where
 
         (
             Self {
-                swarm: Self::build_swarm(key, membership, balancer, monitor, redial_cooldown),
+                swarm: Self::build_swarm(
+                    key,
+                    membership,
+                    balancer,
+                    monitor,
+                    redial_cooldown,
+                    ReplicationConfig {
+                        seen_message_cache_size: replication_settings.seen_message_cache_size,
+                        seen_message_ttl: replication_settings.seen_message_ttl,
```

Comment:
If we really need 2 of those. Better to implement `From/Into` or a custom method `into_replication_config`. But I think there should be just one probably.

Reply:
Another solution would be to to make Replication generic over `Cached` trait from the `cached` crate. Then we'd just pass the cache constructed from the swarm.

Reply:
I prefer single struct config at behavior level, in replication we'll most likely use one implementation for cache, so my suggestion is redundant.

Reply:
> I prefer single struct config at behavior level, in replication we'll most likely use one implementation for cache, so my suggestion is redundant.

:heavy_plus_sign: **1**, I think single struct config is best.